cmake/thirdparty/SetupHIP.cmake

```diff
@@ -9,21 +9,28 @@
 ################################
 # HIP
 ################################
-find_package(hip REQUIRED)
-
-message(STATUS "HIP version:      ${hip_VERSION}")
-message(STATUS "HIP platform:     ${HIP_PLATFORM}")
 
 if (NOT ROCM_PATH)
     find_path(ROCM_PATH
         hip
         ENV{ROCM_DIR}
         ENV{ROCM_PATH}
+        ENV{HIP_PATH}
+        ${HIP_PATH}/..
         ${HIP_ROOT_DIR}/../
         ${ROCM_ROOT_DIR}
         /opt/rocm)
 endif()
 
+# Update CMAKE_PREFIX_PATH to make sure all the configs that hip depends on are
+# found.
+set(CMAKE_PREFIX_PATH "${CMAKE_PREFIX_PATH};${ROCM_PATH}")
+
+find_package(hip REQUIRED CONFIG PATHS ${HIP_PATH} ${ROCM_PATH})
```

Comment:
```suggestion
find_package(hip REQUIRED CONFIG PATHS "${HIP_PATH}" "${ROCM_PATH}")
```

Reply:
Always good practice to quote strings that could be empty since the CMake parsing can give you syntax errors =/